src/target.rs

```diff
@@ -113,4 +116,79 @@ impl Target {
 
         lines
     }
+
+    fn is_debianlike(&self) -> bool {
+        consts::ARCH.eq_ignore_ascii_case(&self.arch)
+            && consts::OS.eq_ignore_ascii_case(&self.os)
+            && PathBuf::from("/etc/debian_version").exists()
+    }
+
+    fn is_freebsd(&self) -> bool {
+        self.os.eq_ignore_ascii_case("freebsd")
+    }
+
+    fn is_haiku(&self) -> bool {
+        self.os.eq_ignore_ascii_case("haiku")
+    }
+
+    fn is_windows(&self) -> bool {
+        self.os.eq_ignore_ascii_case("windows")
+    }
+
+    pub fn default_libdir(&self) -> PathBuf {
+        if self.is_target_overridden {
```

Comment:
It should be `if !is_target_overridden`, then do the debianlike process, run the rest of the checks in any case.